### PR TITLE
Cleanup BMAC logging

### PR DESCRIPTION
**Pull request by carbonin**
- Always use the passed logger
- Don't log entire objects in info messages
- Add agent fields
- Remove unnecessary agent/bmh name and namespaces in messages (use fields)
- Remove logger from functions that don't use it

## List all the issues related to this PR

- [ ] New Feature 
- [x] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [x] Operator Managed Deployments
- [ ] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [ ] No tests needed

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

### CODE REVIEW

File: internal/controller/controllers/bmh_agent_controller.go [hunk 1]
```diff
@@ -679,7 +685,7 @@ func (r *BMACReconciler) reconcileAgentInventory(log logrus.FieldLogger, bmh *bm
 // By re-attaching the BMH and clearing the Image field on it, BMAC will clear
 // the Image data to force the boot from ISO
 func (r *BMACReconciler) reconcileUnboundAgent(log logrus.FieldLogger, bmh *bmh_v1alpha1.BareMetalHost, agent *aiv1beta1.Agent) reconcileResult {
-	log.Debugf("Started Unbound Agent reconcile for agent %s/%s and bmh %s/%s", agent.Namespace, agent.Name, bmh.Namespace, bmh.Name)
+	log.Debugf("Started Unbound Agent reconcile")
```
Comment (avishayt) [inline internal/controller/controllers/bmh_agent_controller.go:688]:
```suggestion
	log.Debugf("Started unbound Aagent reconcile")
```

File: internal/controller/controllers/bmh_agent_controller.go [hunk 2]
```diff
@@ -698,7 +704,7 @@ func (r *BMACReconciler) reconcileUnboundAgent(log logrus.FieldLogger, bmh *bmh_
 	delete(bmh.ObjectMeta.Annotations, BMH_HARDWARE_DETAILS_ANNOTATION)
 	bmh.Spec.Image = nil
 
-	log.Infof("Unbound Agent reconciled to BMH \n %v \n %v", agent, bmh)
+	log.Infof("Unbound Agent reconciled to BMH")
```
Comment (avishayt) [inline internal/controller/controllers/bmh_agent_controller.go:707]:
```suggestion
	log.Infof("Unbound agent reconciled to BMH")
```

File: internal/controller/controllers/bmh_agent_controller.go [hunk 3]
```diff
@@ -882,7 +888,7 @@ func (r *BMACReconciler) reconcileBMH(ctx context.Context, log logrus.FieldLogge
 	bmh.Spec.Image.URL = infraEnv.Status.ISODownloadURL
 	bmh.Spec.Image.DiskFormat = &liveIso
 
-	r.Log.Infof("Image URL has been set in the BareMetalHost  %s/%s", bmh.Namespace, bmh.Name)
+	log.Infof("Image URL has been set in the BareMetalHost")
```
Comment (avishayt) [inline internal/controller/controllers/bmh_agent_controller.go:891]:
```suggestion
	log.Infof("Image URL has been set in the BMH")
```

File: internal/controller/controllers/bmh_agent_controller.go [hunk 4]
```diff
@@ -396,7 +402,7 @@ func (r *BMACReconciler) handleBMHFinalizer(ctx context.Context, log logrus.Fiel
 // the BMH should happen in this reconcile step.
 func (r *BMACReconciler) reconcileAgentSpec(log logrus.FieldLogger, bmh *bmh_v1alpha1.BareMetalHost, agent *aiv1beta1.Agent) reconcileResult {
 
-	log.Debugf("Started Agent Spec reconcile for agent %s/%s and bmh %s/%s", agent.Namespace, agent.Name, bmh.Namespace, bmh.Name)
+	log.Debugf("Started Agent Spec reconcile for agent and bmh")
```
[the remark this answers is not in the document]
Comment (carbonin) [reply internal/controller/controllers/bmh_agent_controller.go:405]: Yes, we set the BMH name and namespace as log fields [here](https://github.com/openshift/assisted-service/blob/ffb5edaa1b4b8d15aa82413749bc9c0fd2ad4f21/internal/controller/controllers/bmh_agent_controller.go#L192-L196)

[the remark this answers is not in the document]
Comment (avishayt) [reply internal/controller/controllers/bmh_agent_controller.go:405]: ACK thanks

**Review by avishayt — APPROVED** [review]

**Pull request merged**